tests/test_methodintervention.py

```diff
@@ -0,0 +1,1180 @@
+    
+    # Set one method to 100%
+    mod.set_method_mix('impl', 1.0)
+    
+    result = mod._build_method_mix_array()
+    
+    assert np.isclose(result.sum(), 1.0)
+    impl_idx = mod.method_mix_order.index('impl')
+    assert np.isclose(result[impl_idx], 1.0)
+    
+    # All others should be ~0
+    for idx, method in enumerate(mod.method_mix_order):
+        if method != 'impl':
+            assert result[idx] < 1e-6
+
+
+def test_method_mix_multiple_methods_fill_100_percent():
+    """Test setting multiple methods that together equal 100%."""
+    mod = fp.MethodIntervention(year=2025)
+    baseline = np.ones(9) / 9.0
+    mod.set_method_mix_baseline(baseline)
+    
+    mod.set_method_mix('pill', 0.4)
+    mod.set_method_mix('inj', 0.3)
+    mod.set_method_mix('impl', 0.3)
+    
+    result = mod._build_method_mix_array()
+    
+    assert np.isclose(result.sum(), 1.0)
+    # All other methods should be ~0
+    for method in ['iud', 'cond', 'btl', 'wdraw', 'othtrad', 'othmod']:
+        idx = mod.method_mix_order.index(method)
+        assert result[idx] < 1e-6
+
+
+def test_method_mix_with_zero_baseline():
+    """Test method mix behavior when some baseline values are zero."""
+    mod = fp.MethodIntervention(year=2025)
+    
+    # Baseline with some zeros
+    baseline = np.array([0.2, 0.2, 0, 0, 0.2, 0, 0.2, 0, 0.2])
+    mod.set_method_mix_baseline(baseline)
+    
+    # Set target for non-zero method
+    mod.set_method_mix('pill', 0.5)
+    
+    result = mod._build_method_mix_array()
+    
+    assert np.isclose(result.sum(), 1.0)
+    pill_idx = mod.method_mix_order.index('pill')
+    assert np.isclose(result[pill_idx], 0.5)
+
+
+def test_method_mix_percentage_conversion():
+    """Test that values >1 are treated as percentages and converted."""
+    mod = fp.MethodIntervention(year=2025)
+    baseline = np.ones(9) / 9.0
+    mod.set_method_mix_baseline(baseline)
+    
+    # Use percentage (not fraction)
+    mod.set_method_mix('impl', 30)  # Should be interpreted as 30% = 0.30
+    
+    result = mod._build_method_mix_array()
+    impl_idx = mod.method_mix_order.index('impl')
+    
+    # Should convert 30 → 0.30
+    assert np.isclose(result[impl_idx], 0.30, atol=0.01)
+
+
+def test_very_small_efficacy():
+    """Test very small but valid efficacy values."""
+    mod = fp.MethodIntervention(year=2025)
+    
+    mod.set_efficacy('wdraw', 0.001)  # 0.1% effective
+    
+    intv = mod.build()
+    assert intv.pars.eff['Withdrawal'] == 0.001
+
+
+def test_very_large_duration():
+    """Test very large duration values (e.g., permanent methods)."""
+    mod = fp.MethodIntervention(year=2025)
+    
+    mod.set_duration_months('btl', 600)  # 50 years (permanent)
+    
+    intv = mod.build()
+    assert intv.pars.dur_use['BTL'] == 600.0
+
+
+def test_fractional_duration():
+    """Test fractional month durations."""
+    mod = fp.MethodIntervention(year=2025)
+    
+    mod.set_duration_months('cond', 0.5)  # Half a month
```

Comment:
Should be able to do things like `mod.set_duration_months('cond', ss.days(1))` as well